### PR TITLE
feature/car-crud

### PR DESCRIPTION
**Pull request by alesiahlushakova**

### CODE REVIEW

File: src/main/java/by/itech/lab/supplier/dto/mapper/CarMapper.java [hunk 1]
```diff
@@ -19,7 +19,7 @@ public Car map(final CarDto dto) {
                 .id(dto.getId())
                 .totalCapacity(dto.getTotalCapacity())
                 .currentCapacity(dto.getCurrentCapacity())
-                .address(addressMapper.map(dto.getAddressDto()))
+                .address(dto.getAddressDto() != null ? addressMapper.map(dto.getAddressDto()) : null)
```
Comment (FortuneAwaker) [inline src/main/java/by/itech/lab/supplier/dto/mapper/CarMapper.java:22]: It is better to use Objects.nonNull

**Review by FortuneAwaker — APPROVED** [review]

File: src/main/java/by/itech/lab/supplier/repository/CarRepository.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+public interface CarRepository extends JpaRepository<Car, Long> {
+
+    @Modifying
+    @Query("update Car set deletedAt = current_timestamp where id = :id")
+    void deleteById(@Param("id") Long id);
+
+    @Modifying
+    @Query("update Car set currentCapacity = currentCapacity - :capacity  where id = :id")
+    int decreaseCurrentCapacity (@Param("capacity") double capacity, @Param("id") Long id);
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/repository/CarRepository.java]: is that method used somewhere?

Comment (svareyko) [reply src/main/java/by/itech/lab/supplier/repository/CarRepository.java]: also, what about usage of only one method with {{currentCapacity + :capacity}} where as capacity put values above zero to increase and below zero to decrease?
for example:
300 + (-10) = 290
300 + (10) = 310

Comment (alesiahlushakova) [reply src/main/java/by/itech/lab/supplier/repository/CarRepository.java]: Can I add TODO with description where to call method in future, when the services that call this method will be developed?

Comment (svareyko) [reply src/main/java/by/itech/lab/supplier/repository/CarRepository.java]: no, in most of the cases there is a really low chance that you will work on that part and most possible that someone who will work on that task will implement his own solution, as he doesn't know that this part is already implemented or he will see the implementation in another way

**Review by IrynaZazyba — APPROVED** [review]

**Pull request merged**